### PR TITLE
Attempt to implement replacement for using select.poll on macosx

### PR DESCRIPTION
**Pull request by fizyk**
With using select.select

### CODE REVIEW

File: src/mirakuru/output.py [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
-
+import platform
 import re
 import select
 from typing import Union, List, Any, TypeVar, Tuple, IO
 
 from mirakuru.base import SimpleExecutor
 
 
+IS_DARWIN = platform.system()
```
Comment (mmaslowskicc) [inline src/mirakuru/output.py]: Missing comparison?

**Comment by mmaslowskicc** [comment]
Can we reliably detect if poll is available? (I remember experiencing some issues with maybe gevent hiding it.)

**Comment by fizyk** [comment]
@mmaslowskicc It appears to be dependand of the compiler used for macos, if it's gcc, then the poll is available, and the simplest test would be to check for the poll function in select module. (we won't be able to test that on travis though - hence I'm looking for the platform name instead)

**Comment by mmaslowskicc** [comment]
Or we could support choosing any implementation at runtime, and tests would check both. Your approach is ok too, it might fix the problem and keeps the code tested.

**Pull request merged**